### PR TITLE
Add locks around Close() and MarkUnusable().

### PR DESCRIPTION
**Pull request by didip**
See: https://github.com/fatih/pool/issues/15

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+import (
+	"net"
+	"sync"
+)
 
 // PoolConn is a wrapper around net.Conn to modify the the behavior of
 // net.Conn's Close() method.
 type PoolConn struct {
 	net.Conn
+	sync.RWMutex
```
Comment (fatih) [inline conn.go]: This will expose the lock outside the package. Can you assign it to an unexported field, such as `mu sync.RWMutex`, would prefer that

Comment (didip) [inline conn.go]: ok, not a problem.

**Comment by didip** [comment]
I have placed the locks under `mu` variable.

**Comment by fatih** [comment]
Thanks @didip

**Pull request merged**